pkg/spec/deploymentconfig.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/ghodss/yaml"
+	os_deploy_v1 "github.com/openshift/origin/pkg/deploy/apis/apps/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Unmarshal the Kedge YAML file
+func (deployment *DeploymentConfigSpecMod) Unmarshal(data []byte) error {
+	err := yaml.Unmarshal(data, &deployment)
+	if err != nil {
+		return errors.Wrap(err, "could not unmarshal into internal struct")
+	}
+	log.Debugf("object unmarshalled: %#v\n", deployment)
+	return nil
+}
+
+// Validate all portions of the file
+func (deployment *DeploymentConfigSpecMod) Validate() error {
+
+	if err := validateVolumeClaims(deployment.VolumeClaims); err != nil {
+		return errors.Wrap(err, "error validating volume claims")
+	}
+
+	return nil
+}
+
+// Fix all services / volume claims / configmaps that are applied
+// TODO: abstract out this code when more controllers are added
+func (deployment *DeploymentConfigSpecMod) Fix() error {
```

Comment:
wrt TODO and what is happening in this method, we can already abstract this like it happens for other controllers -
e.g. the following code from the `Fix()` method for a Deployment -

```golang
func (deployment *DeploymentSpecMod) Fix() error {
	if err := deployment.ControllerFields.fixControllerFields(); err != nil {
		return errors.Wrap(err, "unable to fix ControllerFields")
	}

	deployment.ControllerFields.ObjectMeta.Labels = addKeyValueToMap(appLabelKey, deployment.ControllerFields.Name, deployment.ControllerFields.ObjectMeta.Labels)

	return nil
}

```

Reply:
We cannot use that method now because it has some checks of ingress which cannot be used with OpenShift!

Reply:
@containscafeine also, this TODO was added by you :+1:

Reply:
Tracked in #350